trickle/trickle_server.go

```diff
@@ -435,18 +438,18 @@ func (s *Stream) getForRead(idx int) (*Segment, int, bool) {
 		return seg != nil && seg.idx == i
 	}
 	if idx == -1 {
-		idx = s.latestWrite
```

Comment:
How did this work before if it was "latest" and not "next" and we didn't do +1 here?

Reply:
With preconnects as described in https://github.com/livepeer/go-livepeer/pull/3548#issuecomment-2852141912, "latest" was effectively "next" but hopefully that behavior is less implicit now, with this set of changes.